lib/modules/platform/scm-manager/schema.ts

```diff
@@ -0,0 +1,121 @@
+import { z } from 'zod';
+
+export const UserSchema = z.object({
+  mail: z.string().optional().nullable(),
+  displayName: z.string(),
+  name: z.string(),
+});
+export type User = z.infer<typeof UserSchema>;
+
+export const DefaultBranchSchema = z.object({
+  defaultBranch: z.string(),
+});
+
+export const LinkSchema = z.object({
+  href: z.string(),
+  name: z.string().optional().nullable(),
+  templated: z.boolean().optional().nullable(),
+});
+
+export const LinksSchema = z.record(
+  z.string(),
+  z.union([LinkSchema, z.array(LinkSchema)]),
+);
+export type Link = z.infer<typeof LinkSchema>;
+export type Links = z.infer<typeof LinksSchema>;
```

Comment:
```suggestion
export const Link = z.object({
  href: z.string(),
  name: z.string().optional().nullable(),
  templated: z.boolean().optional().nullable(),
});
export type Link = z.infer<typeof Link>;

export const Links = z.record(
  z.string(),
  z.union([LinkSchema, z.array(Link)]),
);
export type Links = z.infer<typeof Links>;
```

store schema and type together